http-filter-example/http_filter.cc

```diff
@@ -26,9 +29,45 @@ const std::string HttpSampleDecoderFilter::headerValue() const {
   return config_->val();
 }
 
+int HttpSampleDecoderFilter::headerExtra() const {
+  return config_->extra();
+}
+
 FilterHeadersStatus HttpSampleDecoderFilter::decodeHeaders(RequestHeaderMap& headers, bool) {
-  // add a header
-  headers.addCopy(headerKey(), headerValue());
+
+  // parse config: key is header operation, value is a vector of arguments for the operation
+  std::unordered_map<std::string, std::vector<std::string>> header_ops;
+  std::string header_config = headerValue();
+
+  // Find the position of the first space character
+  size_t spacePos = header_config.find(' ');
+  std::string operation = header_config.substr(0, spacePos);
+  std::string args = header_config.substr(spacePos + 1);
+
+  // Create a string stream to iterate over the arguments
+  std::istringstream iss(args);
+  std::vector<std::string> values;
+  std::string value;
+
+  // Split the rest of the string by space and store the values in a vector
+  while (iss >> value) {
+      values.push_back(value);
+  }
+
+  // Insert the key-value pair into header_ops
+  header_ops[operation] = values;
+
+  // perform header operations
+  for (auto it = header_ops.begin(); it != header_ops.end(); ++it) {
```

Comment:
it shoud be like this instead:
```for (auto const& header_op : header_ops)```

Const because you aren't changing the operations inside the loop (rather just applying them). The & to pass by reference and avoid unnecessary copies. Also this is the more modern simpler idiom for looping over a container (a suggestion would be to check the envoy code base itself on what coding style they follow for these things and try to follow that(